examples/common/sample_config.py

```diff
@@ -116,7 +117,26 @@ def create_sample_config(args, parser) -> SampleConfig:
         target_device = sample_config.pop("target_device")
         loaded_json["target_device"] = target_device
     nncf_config = NNCFConfig.from_dict(loaded_json)
+
     if args.disable_compression and 'compression' in nncf_config:
         del nncf_config['compression']
+
     sample_config.nncf_config = nncf_config
     return sample_config
+
+def _fail_if_training_with_eval_only_config(sample_config: SampleConfig):
+    if sample_config.eval_only and 'train' in sample_config.mode:
+        raise RuntimeError(EVAL_ONLY_ERROR_TEXT)
+
+def create_sample_config(args, parser) -> SampleConfig:
+    sample_config = _parse_sample_config(args, parser)
+    _fail_if_training_with_eval_only_config(sample_config)
+    sample_config = _embed_nncf_config(args, sample_config)
+
+    return sample_config
+
+
+def create_sample_config_for_separate_train_eval_scripts(args, parser) -> SampleConfig:
```

Comment:
What different between `create_sample_config` and `create_sample_config_for_separate_train_eval_scripts`? Why you cannot check  `_fail_if_training_with_eval_only_config` everywhere?

Reply:
When train/eval scripts are separate, the train/eval mode is not determined by the `--mode` argument from the CLI (which is then also embedded in the config), but rather it is the choice of the script itself (either `train.py` or `eval.py`) that determines whether we are training or eval-ing.